module/zfs/spa_misc.c

```diff
@@ -444,9 +444,13 @@ spa_config_lock_init(spa_t *spa)
 		spa_config_lock_t *scl = &spa->spa_config_lock[i];
 		mutex_init(&scl->scl_lock, NULL, MUTEX_DEFAULT, NULL);
 		cv_init(&scl->scl_cv, NULL, CV_DEFAULT, NULL);
-		zfs_refcount_create_untracked(&scl->scl_count);
 		scl->scl_writer = NULL;
 		scl->scl_write_wanted = 0;
+#ifdef ZFS_DEBUG
+		zfs_refcount_create_untracked(&scl->scl_count);
+#else
+		scl->scl_count = 0;
+#endif
```

Comment:
Since we always create the refcount as untracked, I don't see the advantage of using a refcount on debug builds.

Reply:
Whoops, just saw the conversation with @behlendorf about this.  If we want to keep the ability to turn on the reference tracking when needed for debugging, I'd prefer to find a way to not litter the code with #ifdef's.  For example, we could have a zfs_intref_t and associated functions like `zfs_intref_add()`, which on debug builds use zfs_refcount_* but on nondebug builds use an `int` and non-atomic ops.  Basically moving the #ifdef uglyness from all call sites to a single header file.